test/distributed/test_c10d_common.py

```diff
@@ -806,6 +806,101 @@ def fut_then(fut):
 
         return fut.then(fut_then)
 
+    def _test_not_nan(self, model, x):
+        y = model(x)
+        self.assertFalse(y.isnan().any().item())
+        y.sum().backward()
+        for p in model.parameters():
+            self.assertFalse(p.grad.isnan().any().item())
+
+    @skip_if_lt_x_gpu(2)
+    def test_sync_batch_norm_only_empty_input(self):
+        pg = self._get_process_group()
+
+        model = torch.nn.Sequential(
+            nn.BatchNorm2d(2),
+        ).to(device=self.rank)
+        model = DistributedDataParallel(
+            model,
+            device_ids=[self.rank],
+            process_group=pg,
+        )
+        model = nn.SyncBatchNorm.convert_sync_batchnorm(
+            model,
+            process_group=pg,
+        )
+
+        model.train()
+
+        # only rank 0 receives empty inputs
+        x = torch.zeros(
+            (1 if self.rank != 0 else 0, 2, 11, 13),
+            dtype=torch.float32,
+            device=self.rank
+        )
+
+        # input requires grad, this will trigger the collective communication
+        # in the backward pass
+        x.requires_grad = True
+        self._test_not_nan(model, x)
+
+        # input does not requires grad
+        x.requires_grad = False
+        self._test_not_nan(model, x)
+
+        # all ranks receive empty inputs
+        x = torch.zeros(
+            (0, 2, 11, 13),
+            dtype=torch.float32,
+            device=self.rank
+        )
+
+        # input requires grad, this will trigger the collective communication
+        # in the backward pass
+        x.requires_grad = True
+        self._test_not_nan(model, x)
+
+        # input does not requires grad
+        x.requires_grad = False
+        self._test_not_nan(model, x)
```

Comment:
@datumbox @albanD is there a way to test the grad value as well?

If I feed batch size 0 and 2 to the two processes, will it generate the same gradients if they receive batch size of 1 and 1 respective of the same data? I assume no, because `invstd` is no longer guaranteed to be the same in this case?

Reply:
I think I agree. It's not going to be the same gradient because the minibatch statistics will be different in the two cases.

Reply:
@datumbox  I don't get why you say "the minibatch statistics will be different" - in SyncBatchNorm the minibatch is __all samples__ on all workers, so in @mrshenli 's example it's the same minibatch (2 samples) in both cases.